include/git2/common.h

```diff
@@ -157,6 +157,7 @@ typedef enum {
 	GIT_OPT_SET_SSL_CERT_LOCATIONS,
 	GIT_OPT_SET_USER_AGENT,
 	GIT_OPT_ENABLE_STRICT_OBJECT_CREATION,
+	GIT_OPT_ENABLE_SYMBOLIC_REF_TARGET_VALIDATION,
```

Comment:
So, I'm afraid that the belated answer here would be no, in my opinion, because we need to be able to resolve refs that have been created without strict validation.

I'm not sure what this means for us presently, given that your fix up renames the constant and has now been merged.